### PR TITLE
Some more cleanups

### PR DESCRIPTION
**Pull request by flatcap**
- da20d060 move functions to library
- 5d9060e0 lib: move MIN/MAX macros
- ddfef085 simplify null checks
- fe4ee20b kill preproc expansion laziness
- ~45a55efe reduce scope of variables~
- 9a1a3b7d fix: variable type when using fread
- d5b3d29e reduce scope of variables

More details in the commit notes.

### CODE REVIEW

File: browser.c [hunk 1]
```diff
@@ -1467,13 +1466,12 @@ void _mutt_select_file(char *f, size_t flen, int flags, char ***files, int *numf
...
         {
           char **tfiles = NULL;
-          int j, k;
 
           if (menu->tagged)
           {
             *numfiles = menu->tagged;
             tfiles = safe_calloc(*numfiles, sizeof(char *));
-            for (j = 0, k = 0; j < state.entrylen; j++)
+            for (int j = 0, k = 0; j < state.entrylen; j++)
```
Comment (gahr) [inline browser.c]: entrylen is `unsigned int`, too

File: buffy.c [hunk 2]
```diff
@@ -89,7 +88,8 @@ static int fseek_last_message(FILE *f)
     bytes_read = fread(buffer, sizeof(char), bytes_read, f);
     if (bytes_read == -1)
```
Comment (gahr) [inline buffy.c]: wow.. this is a bug (that you haven't introduced). `fread` returns `size_t`

File: buffy.c [hunk 3]
```diff
@@ -89,7 +88,8 @@ static int fseek_last_message(FILE *f)
...
       return -1;
-    for (i = bytes_read; --i >= 0;)
+    /* 'i' is Index into `buffer' for scanning.  */
+    for (int i = bytes_read; --i >= 0;)
```
Comment (gahr) [inline buffy.c]: See above, please use `size_t` for `bytes_read`.

**Comment by flatcap** [comment]
I've fixed the `fread()` problem in 9a1a3b7d.

**Pull request closed without merging**